spec/controllers/manage_evaluators_controller_spec.rb

```diff
@@ -84,27 +84,91 @@
       end
     end
 
-    context 'when inviting a new user' do
-      let(:new_user_params) do
-        {
-          evaluator_invitation: {
-            email: 'new_evaluator@example.com',
-            phase_id: phase.id,
-            first_name: 'New',
-            last_name: 'Evaluator',
-            last_invite_sent: Time.current
+    context 'when inviting an existing user with an invitation' do
+      let(:existing_user) { create(:user, role: 'evaluator', status: 'pending') }
+      let!(:invitation) { create(:evaluator_invitation, challenge: challenge, phase: phase, email: existing_user.email) }
+
+      before do
+        ChallengePhasesEvaluator.create!(challenge: challenge, phase: phase, user: existing_user)
+      end
+
+      it 'adds the user as an evaluator without creating a new ChallengePhasesEvaluator' do
```

Comment:
I'm confused which scenario is being tested here. I don't think it should be possible for an existing user to also have an invitation? If an existing user is invited to the challenge, then a ChallengePhaseEvaluator will be created for them and the Invitation should be deleted.. might need to add that delete logic to the controller actually.

the use of `let`, `let!` and `before` blocks makes it hard for me to follow the logical ordering of events in this test. I think this test should be something like "it does not create an EvaluatorInvitation or ChallengePhaseEvaluator when the user is already added to the challenge" and then you can remove the `let!(:invitation)` since there should not be an invitation for an existing user possible.

Reply:
Yeah, I split this logic out - this test is supposed to make sure that an additional ChallengePhaseEvaluator isn't created when that evaluator is already associated with that challenge/phase. I added the same test for an EvaluatorInvitation - an additional EvaluatorInvitation isn't created when an previous invitation for that evaluator and challenge/phase exists. 

Do I need to add the delete logic to the controller if the deleting the invitation is handled as an after_create on the user already? 

```
class User
  after_create :accept_evaluator_invitation
  
  private
  def accept_evaluator_invitation
    EvaluatorInvitation.where(email: self.email).each do |invite|
      ChallengePhasesEvaluator.create(challenge: invite.challenge, phase: invite.phase, user: self)
      invite.destroy
    end
  end
end
```

Reply:
I think you're right 👍